### PR TITLE
Refactor Monitor Metric Aggregation

### PR DESCRIPTION
**Pull request by pan-x-c**
## Description

This pull request introduces a unified, extensible system for metrics aggregation across the codebase, replacing ad-hoc implementations with a new `trinity.utils.metrics` module. The new system supports per-metric aggregation types via key suffixes (e.g., `reward:mean`, `tokens:sum`) and ensures consistent metric handling for training, evaluation, and monitoring workflows. It also updates all relevant aggregation call sites and adds comprehensive tests for the new utilities.

**Core metrics aggregation system:**

* Added `trinity/utils/metrics.py`, which defines a unified metrics aggregation API (`aggregate_metrics`, `aggregate_eval_metrics`, `aggregate_run_level_metrics`) supporting aggregation-type suffixes in metric keys and providing flexible output statistics. The module also includes a parser for metric keys and an `AggType` enum.

* Added thorough tests for the new metrics utilities in `tests/utils/test_metrics.py`, covering key parsing, aggregation logic, compatibility, and edge cases.

**Refactoring and integration:**

* Refactored `trinity/explorer/scheduler.py` to use the new metrics aggregation system in `calculate_task_level_metrics`, supporting aggregation-type suffixes and proper handling for both training and evaluation tasks. [[1]](diffhunk://#diff-367903307c678e9b65dccc0de2dea57feef110bba4c5d7935b0467ac57f4ac4fL101-R167) [[2]](diffhunk://#diff-367903307c678e9b65dccc0de2dea57feef110bba4c5d7935b0467ac57f4ac4fL143-L146)

* Refactored `trinity/explorer/workflow_runner.py` to use `aggregate_run_level_metrics` for experience-to-run metrics aggregation, ensuring suffix-aware and consistent behavior. [[1]](diffhunk://#diff-63c341c80e795628b7fddc34251eaf4870f2f14a95afb164f4f38754e789fd8fL8) [[2]](diffhunk://#diff-63c341c80e795628b7fddc34251eaf4870f2f14a95afb164f4f38754e789fd8fL48-R57)

* Updated `trinity/utils/monitor.py` to use the new aggregation utilities (`aggregate_metrics`, `aggregate_eval_metrics`) in `gather_metrics` and `gather_eval_metrics`, replacing previous pandas-based logic. [[1]](diffhunk://#diff-0cae247e9b4de1881e9743d2a8629e9f7dcf506f2e768825ec1f34b9d4246cdfR45-R50) [[2]](diffhunk://#diff-0cae247e9b4de1881e9743d2a8629e9f7dcf506f2e768825ec1f34b9d4246cdfR61-R68)

**Other:**

* Commented out the `[tool.uv.extra-build-dependencies]` section in `pyproject.toml` for now, as part of project maintenance.

## Checklist

Please check the following items before code is ready to be reviewed.

- [ ]  Code has passed all tests
- [ ]  Docstrings have been added/updated in Google Style
- [ ]  Documentation has been updated
- [ ]  Code is ready for review

### CODE REVIEW

File: trinity/explorer/scheduler.py [hunk 1]
```diff
@@ -97,53 +97,98 @@ def bootstrap_metric(
...
-                    result[f"{key}/best@{n}"] = bon_mean
-                    result[f"{key}/worst@{n}"] = won_mean
+            elif agg == AggType.SUM:
+                result[key] = sum(values)
+            elif agg == AggType.MAX:
+                result[key] = max(values)
+            elif agg == AggType.MIN:
+                result[key] = min(values)
+            elif agg == AggType.LAST:
+                result[key] = values[-1]
```
Comment (chenyushuo) [inline trinity/explorer/scheduler.py]: It is recommended to add a helper function to handle this logic, and the same applies to the following

File: trinity/utils/metrics.py [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+
+import numpy as np
+
+
+class AggType(str, Enum):
+    MEAN = "mean"
+    SUM = "sum"
+    MAX = "max"
+    MIN = "min"
+    LAST = "last"
```
Comment (chenyushuo) [inline trinity/utils/metrics.py:20]: may add to constants.py ?

File: trinity/utils/metrics.py [hunk 3]
```diff
@@ -0,0 +1,205 @@
...
+                if fn is not None:
+                    result[f"{prefix_str}{name}/{stat}"] = fn(arr).item()
+        elif agg == AggType.SUM:
+            result[f"{prefix_str}{name}/sum"] = sum(values)
+        elif agg == AggType.MAX:
+            result[f"{prefix_str}{name}/max"] = max(values)
+        elif agg == AggType.MIN:
+            result[f"{prefix_str}{name}/min"] = min(values)
+        elif agg == AggType.LAST:
+            result[f"{prefix_str}{name}"] = values[-1]
```
Comment (chenyushuo) [inline trinity/utils/metrics.py]:
```suggestion
            result[f"{prefix_str}{name}/last"] = values[-1]
```

File: trinity/utils/metrics.py [hunk 4]
```diff
@@ -0,0 +1,205 @@
...
+            arr = np.array(values)
+            if detailed_stats:
+                stat_fns = {
+                    "mean": np.mean,
+                    "max": np.max,
+                    "min": np.min,
+                    "std": np.std,
+                }
+                for stat in output_stats:
+                    fn = stat_fns.get(stat)
```
Comment (chenyushuo) [inline trinity/utils/metrics.py]:
```suggestion
                    fn = getattr(np, stat, None)
```

File: trinity/utils/metrics.py [hunk 5]
```diff
@@ -0,0 +1,205 @@
...
+            else:
+                result[f"{prefix_str}{name}"] = np.mean(arr).item()
+        elif agg == AggType.SUM:
+            result[f"{prefix_str}{name}/sum"] = sum(values)
+        elif agg == AggType.MAX:
+            result[f"{prefix_str}{name}/max"] = max(values)
+        elif agg == AggType.MIN:
+            result[f"{prefix_str}{name}/min"] = min(values)
+        elif agg == AggType.LAST:
+            result[f"{prefix_str}{name}"] = values[-1]
```
Comment (chenyushuo) [inline trinity/utils/metrics.py]:
```suggestion
            result[f"{prefix_str}{name}/last"] = values[-1]
```

**Review by chenyushuo — APPROVED** [review]
